### PR TITLE
chore: change type name of AsyncNode to Node

### PR DESCRIPTION
**Pull request by fregataa**
React compiler use the name of type whether it is `Node` or not.
It raises error if the name is not `Node`.

**Checklist:** (if applicable)

- [x] Milestone metadata specifying the target backport version

### CODE REVIEW

File: changes/1757.fix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Change type name of AsyncNode to Node since React compiler use it to determine relay node.
```
Comment (rapsealk) [inline changes/1757.fix.md]:
```suggestion
Change type name of AsyncNode to Node since Relay compiler use it to determine relay node.
```

**Review by rapsealk — COMMENTED** [review]
Isn't is `Relay` compiler that throws an error?

**Comment by rapsealk** [comment]
Tested with:
```typescript title="ModelCardModal.tsx"
// ModelCardModal.tsx
  const model_card = useFragment(
    graphql`
      fragment ModelCardModalFragment on ModelCard
      @refetchable(queryName: "ModelCardModalRefetchQuery") {
        id
        name
      }
    `,
    modelCardModalFrgmt,
  );
```

Refs: https://relay.dev/graphql/objectidentification.htm#sec-Node-Interface

**Review by rapsealk — APPROVED** [review]
LGTM

**Review by yomybaby — APPROVED** [review]
LGTM 🚀 Thank you.

**Pull request merged**